src/libraries/__test__/extendComponentProps.js

```diff
@@ -0,0 +1,53 @@
+import extendComponentProps from '../extendComponentProps';
+import {Map} from 'immutable';
+import test from 'ava';
+
+test('extend components props', t => {
+  const builtProps = Map({
+    children: 'Any children',
+    decorated: 'Test',
+    loading: false
+  })
+  const propsDefinition = Map({
```

Comment:
You can just call `fromJS` on an object and it will get converted to nested Maps/Lists

```
const propsDefinition = fromJS({children: {description: ''}, ...}
```